src/search1_conversion/convert_result.py

```diff
@@ -1,22 +1,55 @@
-import re
 from src.utils.config import config
-from src.utils.formatting import iso8601_to_epoch
+from src.utils.formatting import iso8601_to_epoch_ms
 from src.utils.user_profiles import get_user_profiles
 from src.utils.workspace import get_workspace_info
+from src.exceptions import NoAccessGroupError, NoUserProfileError
 
-# Mappings from search2 document fields to search1 fields:
-_KEY_MAPPING = {
+# TODO: The structure of the ES docs and of the API's result
+# data should be documented in detail outside of this file, with a
+# reference here.
+
+# Mappings from search2 document fields to search1 fields.
+_GLOBAL_DOC_KEY_MAPPING = {
     'obj_name': 'object_name',
-    'access_group': 'access_group',
-    'obj_id': 'obj_id',
-    'version': 'version',
-    'timestamp': 'timestamp',
-    'obj_type_name': 'type',
-    'creator': 'creator'
+    'access_group': 'workspace_id',
+    'obj_id': 'object_id',
+    'version': 'object_version',
+    'obj_type_module': 'workspace_type_module',
+    'obj_type_name': 'workspace_type_name',
+    'obj_type_version': 'workspace_type_version',
+    'timestamp': 'modified_at'
 }
 
+# These keys are copied over literally without renaming
+# keys or transformation
+_GLOBAL_DOC_KEY_COPYING = [
+    'creator',
+    'copied'
+]
+
+# These keys are copied from the result "hit", not "hit.doc" as
+# above.
+_GLOBAL_HIT_KEY_COPYING = [
+    'id'
+]
+
+# These keys are to be neither mapped nor copied, but when copying the rest of the
+# doc fields into the data field, should be excluded, or omitted.
+_GLOBAL_DOC_KEY_EXCLUSION = [
+    'is_public',
```

Comment:
Are you dropping this just because it may not be in sync?

Reply:
This is just copying the field into the api, it is still in the index.
And, right, this can get out of sync for most if not all objects other than Narratives. 
Search results can optionally return the workspace info, which gives the true public permission value.